src/__tests__/parseWithPointers.spec.ts

```diff
@@ -514,4 +514,126 @@ european-cities: &cities
       expect(result.diagnostics).toEqual([]);
     });
   });
+
+  describe('keys sorting', () => {
+    it('does not retain the order of keys by default', () => {
+      const { data } = parseWithPointers(`foo: true
+bar: false
+"1": false
+"0": true
+`);
+
+      expect(Object.keys(data)).toEqual(['0', '1', 'foo', 'bar']);
+    });
+
+    describe('when sortKeys option is set to true', () => {
+      it('retains the initial order of keys', () => {
+        const { data } = parseWithPointers(
+          `foo: true
+bar: false
+"1": false
+"0": true
+`,
+          { sortKeys: true },
+        );
+
+        expect(Object.keys(data)).toEqual(['foo', 'bar', '1', '0']);
+      });
+
+      it('handles duplicate properties', () => {
+        const { data } = parseWithPointers(
+          `{
+      foo: true,
+      bar: false,
+      "0": 0,
+      foo: null,
+      "1": false,
+      "0": true,
+      "1": 0,
+    }`,
+          { sortKeys: true },
+        );
+
+        expect(Object.keys(data)).toEqual(['bar', 'foo', '0', '1']);
```

Comment:
Should `foo` not be first? Since it appears first in the data? `[foo, bar, 0, 1]`

Reply:
It in fact appears first *but* foo is overwritten at line 549, therefore foo and bar keys are swapped.

Reply:
Ah OK so it's last time it's included, not first time, that determines ordering.

Reply:
> Ah OK so it's last time it's included, not first time, that determines ordering.

Yeah, exactly